hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconContext.java

```diff
@@ -0,0 +1,140 @@
+  private Map<ErrorCode, List<String>> errCodeImpactMap = new HashMap<>();
+  private List<ErrorCode> errors = new ArrayList<>();
+  private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+  @Inject
+  public ReconContext(OzoneConfiguration configuration, ReconUtils reconUtils) {
+    this.reconUtils = reconUtils;
+    this.ozoneConfiguration = configuration;
+    threadNamePrefix = reconUtils.getReconNodeDetails(configuration).threadNamePrefix();
+    initializeErrCodeMetaData();
+  }
+
+  private void initializeErrCodeMetaData() {
+    // This maps the error code to error message.
+    errCodeMsgMap.put(ErrorCode.OK, "Recon is healthy !!!");
+    errCodeMsgMap.put(ErrorCode.INVALID_NETWORK_TOPOLOGY,
+        "Invalid network topology of datanodes. Failed to register and load datanodes. Pipelines may not be " +
+            "healthy !!!");
+    errCodeMsgMap.put(ErrorCode.CERTIFICATE_INIT_FAILED, "Error during initializing Recon certificate !!!");
+    errCodeMsgMap.put(ErrorCode.INTERNAL_ERROR,
+        "Unexpected internal error. Kindly refer ozone-recon.log file for details !!!");
+    errCodeMsgMap.put(ErrorCode.GET_OM_DB_SNAPSHOT_FAILED, "OM DB Snapshot sync failed !!!");
+
+    // This maps the error codes to impacted areas of Recon.
+    errCodeImpactMap.put(ErrorCode.INVALID_NETWORK_TOPOLOGY, Arrays.asList("Datanodes", "Pipelines"));
+    errCodeImpactMap.put(ErrorCode.CERTIFICATE_INIT_FAILED, Arrays.asList("Initializing secure Recon"));
+    errCodeImpactMap.put(ErrorCode.INTERNAL_ERROR, Arrays.asList("Recon health"));
+    errCodeImpactMap.put(ErrorCode.GET_OM_DB_SNAPSHOT_FAILED, Arrays.asList("Overview (OM Data)", "OM DB Insights"));
+  }
+
+  /**
+   * @param healthStatus : update Health Status of Recon.
+   */
+  public void updateHealthStatus(boolean healthStatus) {
+    lock.writeLock().lock();
+    try {
+      LOG.info("Update healthStatus of Recon from {} to {}.", isHealthy, healthStatus);
+      this.isHealthy = healthStatus;
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  public boolean isHealthy() {
+    lock.readLock().lock();
+    try {
+      return this.isHealthy;
+    } finally {
+      lock.readLock().unlock();
+    }
+  }
+
+  public String threadNamePrefix() {
+    return threadNamePrefix;
+  }
+
+  public Map<ErrorCode, String> getErrCodeMsgMap() {
+    return errCodeMsgMap;
+  }
+
+  public Map<ErrorCode, List<String>> getErrCodeImpactMap() {
+    return errCodeImpactMap;
+  }
+
+  public List<ErrorCode> getErrors() {
+    return errors;
+  }
+
+  public void updateErrors(ErrorCode errorCode) {
+    lock.writeLock().lock();
+    try {
+      errors.add(errorCode);
+    } finally {
+      lock.writeLock().unlock();
+    }
+  }
+
+  /**
+   * Error codes to make it easy to decode these errors in Recon.
+   */
+  public enum ErrorCode {
+    OK,
+    INVALID_NETWORK_TOPOLOGY,
+    CERTIFICATE_INIT_FAILED,
+    INTERNAL_ERROR,
+    GET_OM_DB_SNAPSHOT_FAILED
+  }
```

Comment:
**Would it be a good idea to use an enum constructor like this? that will encapsulate error messages and impacts directly within the enum, which can improve maintainability and readability.**
```
public enum ErrorCode {
    OK("Recon is healthy !!!", Collections.emptyList()),
    INVALID_NETWORK_TOPOLOGY(
        "Invalid network topology of datanodes. Failed to register and load datanodes. Pipelines may not be healthy !!!",
        Arrays.asList("Datanodes", "Pipelines")),
    CERTIFICATE_INIT_FAILED(
        "Error during initializing Recon certificate !!!",
        Arrays.asList("Initializing secure Recon")),
    INTERNAL_ERROR(
        "Unexpected internal error. Kindly refer ozone-recon.log file for details !!!",
        Arrays.asList("Recon health")),
    GET_OM_DB_SNAPSHOT_FAILED(
        "OM DB Snapshot sync failed !!!",
        Arrays.asList("Overview (OM Data)", "OM DB Insights"));

    private final String message;
    private final List<String> impacts;

    ErrorCode(String message, List<String> impacts) {
        this.message = message;
        this.impacts = impacts;
    }

    public String getMessage() {
        return message;
    }

    public List<String> getImpacts() {
        return impacts;
    }
}

```
### Modification in ReconContext:

```
private void initializeErrCodeMetaData() {
    for (ErrorCode errorCode : ErrorCode.values()) {
        errCodeMsgMap.put(errorCode, errorCode.getMessage());
        errCodeImpactMap.put(errorCode, errorCode.getImpacts());
    }
}
```

Reply:
Thanks for reviewing @ArafatKhan2198 . Yes it is a good idea to make it concise. Will do changes as per your suggestion.